lib/ecstatic.js

```diff
@@ -330,6 +339,19 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
+      isGzip = false;
+
+  stream.on("close", function () {
+    cb(isGzip);
+  });
+
+  stream.on("data", function (chunk) {
```

Comment:
Ignoring the error is intentional

* node API mentions If autoClose is set to true (default behavior), on error or end the file descriptor will be closed automatically. (I should remove `{ autoClose: true }`). if an `error` is emitted, `close` will be emitted too because of `{ autoClose: true }`

* `buffer[0] == 31 && buffer[1] == 139);` is still safe and no need to the length check even if the buffer is empty. It saves some operations when the file is a real gzip.

* As for the `cb` with `err` argument, if there is an error, I think the cb does not need to include `err` because the result must be `cb(false)`. Meanwhile, `statFile()` will finally handle the error.

* Buffer('') resolves the 0.12 issue.